### DISCUSSION TITLE
Can I receive back refresh and router push, and link events?

### DISCUSSION THREAD

**Discussion by sanghogu** [vercel/next.js#59406] [App Router]
I have hundreds of pages now and there is a major flaw in the page. It takes a few seconds for the page to load for the first time and the page appears to be frozen.
cannot reduce the page loading time.
So I'm trying to display a loading screen when moving the page.
help me

**Comment by matthewraaff** [comment]
Hi, you can use [loading.js](https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming) for loading screens.
